kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaHistoryManager.java

```diff
@@ -162,6 +167,32 @@ public static long getEarliestRecreatableCommit(Engine engine, Path logPath)
         }
     }
 
+    /** Get the persisted commit info (if available) for the given delta file. */
+    public static Optional<CommitInfo> getCommitInfoOpt(
+            Engine engine,
+            Path logPath,
+            long version) throws IOException {
+        final FileStatus file = listFrom(engine, logPath, version).next();
```

Comment:
This is prone to failure, if the target file version (hence the file name) is already known --- the file status retrieved directly. If the API for retrieving a filestatus for a file name is not available, you should explicitly extract the needed file from the listing.